### ISSUE TITLE
Badge Request: GitHub Check ignore pending builds

### ISSUE THREAD

**Issue by oliviertassinari** [badges/shields#10221]
### 📋 Description

I have migrated to https://shields.io/badges/git-hub-branch-status, but compared to the CircleCI badge, it shows the latest build status: <img width="108" alt="SCR-20240602-qbse" src="https://github.com/badges/shields/assets/3165635/55e2af73-b73e-46e1-9d17-0b38b1c2adf6">

while the CircleCI badge shows the latest non-pending build status. CircleCI seems the expected behavior. The use case is to be able to, with a glance, see if there is something to be worried about (broken CI branch).

It would be great if we could fix this.

### 🔗 Data

-

### 🎤 Motivation

-

**Comment by PyvesB** [comment]
Hello @oliviertassinari 👋🏻 

Our GitHub branch status badge directly forwards the status returned by the GitHub API (see [code](https://github.com/badges/shields/blob/0360ba3a1b193b25a03454e36b823aa20155d7fc/services/github/github-checks-status.service.js#L73-L79)) . I believe that this is the same status as the one shown in the GitHub UI (green check, red cross, orange dot). In addition to changing a historical behaviour which users seem happy with (this badge was implemented all the way back in #5973), I feel that introducing an inconsistency and having the badge show something different than the GitHub UI would be misleading and confusing.

**Comment by oliviertassinari** [comment]
> I feel that introducing an inconsistency and having the badge show something different than the GitHub UI would be misleading and confusing.

@PyvesB I agree, these are fundamentally two different concepts. The bulk of this issue is to request the introduction of another concept that I would expect to match the most common use case for this badge.

**Comment by PyvesB** [comment]
Nowadays, we have the [GitHub Check Runs badge](https://shields.io/badges/git-hub-branch-check-runs), which I believe should cover this case. Will go ahead and close this issue, feel free to open a new issue if you're still encountering issue with how we display statuses. :)
